### PR TITLE
bsunanda:Run2-alca34 Update Pythia Isolated Track Filter

### PR DESCRIPTION
**Pull request by bsunanda**
useful to create sample of QCD MC events with isolated tracks

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test
________________________________

**Comment by slava77** [comment]
@bsunanda thanks for the cleanup of clang warnings.
Note, however, that the error reported in https://github.com/cms-sw/cmssw/pull/13248#issuecomment-183297146 was due to a segfault during compilation.

**Comment by bsunanda** [comment]
@cmsbuild Please test
These changes affect only one generator level filter - so the error has nothing to do with this PR.

**Comment by bsunanda** [comment]
@cmsbuild Please test
Try once more (see if the spurious failure can be avoided)

**Comment by bsunanda** [comment]
@cmsbuild Please test
These changes affect only one generator level filter - so the error has nothing to do with this PR.

**Comment by bsunanda** [comment]
@vciulli, @covarell, @thuer, @bendavid, @davidlange6, @govoni can you please review it and eventually sign? The unit test failure has nothing to do with this PR. Thanks

File: GeneratorInterface/GenFilters/src/PythiaFilterIsolatedTrack.cc [hunk 1]
```diff
@@ -1,63 +1,58 @@
...
   double deltaPhi;
   double etaEC=100;
   double phiEC=100;
   double Rcurv=pT*33.3*100/38; //r(m)=pT(GeV)*33.3/B(kG)
-  double ecDist=317;  //distance to ECAL andcap from IP (cm), 317 - ecal (not preshower), preshower -300
-  double ecRad=129;  //radius of ECAL barrel (cm)
   double theta=2*atan(exp(-etaIP));
   double zNew;
-  if (theta>0.5*acos(-1)) theta=acos(-1)-theta;
+  if (theta>CLHEP::halfpi) theta=CLHEP::pi-theta;
```
Comment (slava77) [inline GeneratorInterface/GenFilters/src/PythiaFilterIsolatedTrack.cc]: BTW, there are system-level M_PI (```#include <cmath> ``` should bring it in)

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/13248	 280b7fc
- reco code changes are simple, cleaning up the clang warnings
- jenkins tests pass (ignoring the occasionally failing unit test, unrelated to this PR) and comparisons with baseline show no differences

**Comment by bsunanda** [comment]
@vciulli @covarell @thuer @bendavid @davidlange6 @govoni can you please review it and eventually sign? The unit test failure has nothing to do with this PR. Thanks

**Comment by covarell** [comment]
+1

**Comment by bsunanda** [comment]
It is integrated with 8_1_X and backported to 7_1_X

**Pull request closed without merging**